### PR TITLE
Use Thread BR extended address in WS API calls

### PR DESCRIPTION
**Pull request by agners**
## Proposed change

This follows the HA Core change and passes the extended address to the OTBR WS API calls where necessary. It also follows the new OTBR info format which potentially includes multiple OTBRs.

This allows to support multiple OTBR managed by a single system.

Note: There is one corner case when none of the OTBR is found via discovery. In this case we offer to reset the OTBR. Currently we simply offer this for the primary or first one found.

Related Core PR: https://github.com/home-assistant/core/pull/108282

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Enhanced Thread network configuration with added functionality for handling OTBR information more comprehensively.

- **Bug Fixes**
  - Improved handling of extended addresses in network creation, setting, and channel configuration.

- **Documentation**
  - Updated translation strings for clearer descriptions of Thread network actions.

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts [hunk 1]
```diff
@@ -160,19 +161,24 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
   }
 
   private _renderNetwork(network: ThreadNetwork) {
+    const otbrForNetwork =
+      this._otbrInfo &&
+      network.dataset &&
+      ((network.dataset.preferred_extended_address &&
+        this._otbrInfo[network.dataset.preferred_extended_address]) ||
+        Object.values(this._otbrInfo).find(
+          (otbr) => otbr.extended_pan_id === network.dataset?.extended_pan_id
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts:170]:
```suggestion
          (otbr) => otbr.extended_pan_id === network.dataset!.extended_pan_id
```

File: src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts [hunk 2]
```diff
@@ -313,31 +319,33 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
...
             >
           </div>`
         : ""}
     </ha-card>`;
   }
 
-  private _sendCredentials() {
+  private _sendCredentials(ev) {
+    const otbr = (ev.currentTarget as any).otbr as OTBRInfo;
     if (!this._otbrInfo) {
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts:332]:
```suggestion
    if (!otbr) {
```

File: src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts (hunk 1 above)
Comment (agners) [reply src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts]: I am kinda surprised that the linter wouldn't let me use `network.dataset.extended_pan_id` here. I clearly check for `network.dataset` above :man_shrugging:

File: src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts [hunk 3]
```diff
@@ -463,18 +472,23 @@
...
     }
   }
 
-  private async _resetBorderRouter() {
+  private _resetBorderRouterEvent(ev) {
+    const otbr = (ev.currentTarget as any).otbr as OTBRInfo;
+    this._resetBorderRouter(otbr);
+  }
+
+  private async _resetBorderRouter(otbr: OTBRInfo) {
```
[the remark this answers is not in the document]
Comment (agners) [reply src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts:491]: We use the same pattern in other places.

[the remark this answers is not in the document]
Comment (agners) [reply src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts:491]: I am happy with what we have.

File: src/data/otbr.ts [hunk 4]
```diff
@@ -5,33 +5,46 @@ export interface OTBRInfo {
   border_agent_id: string;
   channel: number;
   extended_address: string;
+  extended_pan_id: string;
   url: string;
 }
 
-export const getOTBRInfo = (hass: HomeAssistant): Promise<OTBRInfo> =>
+export type OTBRInfoDict = {
+  [key: string]: OTBRInfo;
```
[the remark this answers is not in the document]
Comment (agners) [reply src/data/otbr.ts:13]: Done.

File: src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts [hunk 5]
```diff
@@ -160,25 +161,36 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
...
+                  "ui.panel.config.thread.thread_network_info"
+                )}
+                .otbr=${otbrForNetwork}
                 .network=${network}
                 .path=${mdiInformationOutline}
                 @click=${this._showDatasetInfo}
               ></ha-icon-button
               >${!network.dataset.preferred && !network.routers?.length
                 ? html`<ha-icon-button
+                    label=${this.hass.localize(
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts:191]:
```suggestion
                    .label=${this.hass.localize(
```

File: src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts [hunk 6]
```diff
@@ -160,25 +161,36 @@ export class ThreadConfigPanel extends SubscribeMixin(LitElement) {
...
-        network.dataset.extended_pan_id
-      );
+      otbrForNetwork;
 
     return html`<ha-card>
       <div class="card-header">
         ${network.name}${network.dataset
           ? html`<div>
               <ha-icon-button
+                label=${this.hass.localize(
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/thread/thread-config-panel.ts:181]:
```suggestion
                .label=${this.hass.localize(
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**